### DISCUSSION TITLE
Unable to access node module in Flow (cannot find module 'theModule')

### DISCUSSION THREAD

**Discussion by smurflo2** [directus/directus#18046] [Archive]
### Describe the Bug

I am running Directus via Docker and I'm trying to use the openai npm package in a Flow. I have set FLOWS_EXEC_ALLOWED_MODULES: 'openai' in my docker compose file, but when I actually try to use the package in a Run Script operation it errors with "Cannot find module 'openai'"

What am I doing wrong here?

### To Reproduce

`docker-compose.yml`:
![image](https://user-images.githubusercontent.com/63376352/229639376-8dd42d43-a928-4218-8d7a-f2ab96eee681.png)

The Flow: It's triggered by a webhook and then I have it pointing to a Run Script operation. The Run Script operation claims that the openai package can be used
![image](https://user-images.githubusercontent.com/63376352/229639807-b1667f82-670e-4520-b6db-c1c0db027eb7.png)

... but when the flow actually runs I get an error: "Cannot find module 'openai'"
![image](https://user-images.githubusercontent.com/63376352/229640211-c8ef4c07-272d-4094-8766-f53292d1c179.png)

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by rijkvanzanten** [comment]
The FLOWS_EXEC_ALLOWED_MODULES will allow the script to try to import the module, however, it doesn't automatically install those modules. In this case, make sure to install the `openai` module as well 👍🏻

**Reply by smurflo2** [reply]
Thanks! That makes sense. Can I get some help on how to install a node module? 🙏

I tried creating a Dockerfile which looks something like:
```
FROM directus/directus:latest
RUN npm install openai
```

and then I changed the line for the directus service in docker-compose.yml from
`image: directus/directus:latest`
to
`build: .`

But that just results in "directus exited with code 0" when I run `docker compose up`

**Reply by smurflo2** [reply]
Finally got it to work with this! (no Dockerfile, just added a command to docker-compose.yml)
![image](https://user-images.githubusercontent.com/63376352/229879763-c9821f2c-a28f-4a94-aa09-1f80de62cec6.png)
